### PR TITLE
Adds custom monthly amount to the list

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/2152

Native implementation: https://github.com/brave-intl/bat-native-ledger/pull/181

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
defined in https://github.com/brave/brave-browser/issues/2152 (for developers)
for QA:
- enable rewards
- change monthly amount
- reload page
- make sure that amount is preserved
- add some sites to ac table
- make contribution
- make sure that correct amount was used

## Reviewer Checklist:

- [x] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [x] Adequate test coverage exists to prevent regressions 
- [x] Verify test plan is specified in PR before merging to source

### CODE REVIEW

**Comment by bsclifton** [comment]
@NejcZdovc here is the patch I tested with:
```diff
diff --git a/components/brave_rewards/browser/rewards_service_impl.cc b/components/brave_rewards/browser/rewards_service_impl.cc
index 5307746f..0f042cee 100644
--- a/components/brave_rewards/browser/rewards_service_impl.cc
+++ b/components/brave_rewards/browser/rewards_service_impl.cc
@@ -1085,6 +1085,8 @@ void RewardsServiceImpl::SetUserChangedContribution() const {
 }

 void RewardsServiceImpl::SetAutoContribute(bool enabled) const {
+  SetUserChangedContribution();
+  SetContributionAmount(13);
   return ledger_->SetAutoContribute(enabled);
 }
```

I then did the following:
1. Fresh profile
2. Enable rewards
3. Toggle auto-contribute off and then on again
4. Value properly displays as `13 BAT` 👍

File: browser/ui/webui/brave_rewards_ui.cc [hunk 1]
```diff
@@ -298,6 +298,8 @@ void RewardsDOMHandler::OnWalletProperties(
     }
 
     result.SetDictionary("wallet", std::move(walletInfo));
+    result.SetDouble("monthlyAmount", wallet_properties->montly_amount);
```
Comment (bsclifton) [inline browser/ui/webui/brave_rewards_ui.cc]: per comment from @bbondy, could a test be added here to ensure `monthlyAmount` is set?

File: components/brave_rewards/browser/rewards_service_impl.cc [hunk 2]
```diff
@@ -533,6 +533,7 @@ void RewardsServiceImpl::OnWalletInitialized(ledger::Result result) {
 
 void RewardsServiceImpl::OnWalletProperties(ledger::Result result,
     std::unique_ptr<ledger::WalletInfo> wallet_info) {
+  LOG(ERROR) << "NEJC - 1";
```
Comment (bsclifton) [inline components/brave_rewards/browser/rewards_service_impl.cc]: todo: remove

File: components/brave_rewards/browser/rewards_service_impl.cc [hunk 3]
```diff
@@ -929,7 +930,12 @@ void RewardsServiceImpl::TriggerOnWalletInitialized(int error_code) {
 void RewardsServiceImpl::TriggerOnWalletProperties(int error_code,
     std::unique_ptr<ledger::WalletInfo> wallet_info) {
   std::unique_ptr<brave_rewards::WalletProperties> wallet_properties;
+
+  LOG(ERROR) << "NEJC - 2";
```
Comment (bsclifton) [inline components/brave_rewards/browser/rewards_service_impl.cc]: todo: remove

File: components/brave_rewards/browser/rewards_service_impl.cc [hunk 4]
```diff
@@ -929,7 +930,12 @@ void RewardsServiceImpl::TriggerOnWalletInitialized(int error_code) {
...
+
   for (auto& observer : observers_) {
+
+  LOG(ERROR) << "NEJC - 3";
```
Comment (bsclifton) [inline components/brave_rewards/browser/rewards_service_impl.cc]: todo: remove

**Comment by bsclifton** [comment]
Getting some unit test failures on this... will need to inspect:
```
2 tests failed:
    RapporLogUploaderTest.NoFetch (../../brave/components/rappor/log_uploader_unittest.cc:60)
    RewardsServiceTest.OnWalletProperties2 (../../brave/components/brave_rewards/browser/rewards_service_impl_unittest.cc:165)
```

**Comment by bbondy** [comment]
Was there an approval request for uplifting this to 0.57.x?

**Comment by bbondy** [comment]
This is breaking builds on Windows

```
../../brave/vendor/bat-native-ledger/src/bat_state.cc(119,10):  error: no member named 'sort' in namespace 'std'
    std::sort(properties.parameters_choices_.begin(),
    ~~~~~^
../../brave/vendor/bat-native-ledger/src/bat_state.cc(281,12):  error: no member named 'sort' in namespace 'std'
      std::sort(properties.parameters_choices_.begin(),
      ~~~~~^
2 errors generated.
```

**Comment by bbondy** [comment]
Uplift approval process is documented here:
https://github.com/brave/brave-browser/wiki/Triage-Guidelines#uplifting

**Pull request merged**